### ISSUE TITLE
SC2216: false positive: piping to `nice` is allowed if it is running a command that uses stdin

### ISSUE THREAD

**Issue by pabs3** [koalaman/shellcheck#1313]
#### For bugs

- Rule Id (if any, e.g. SC1000): SC2216
- My shellcheck version (`shellcheck --version` or "online"): 0.4.7
- [x] I could not try on shellcheck.net because it requires JavaScript

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
find . -type f -print0 | xargs --null cat -- | nice -15 cat
```

#### Here's what shellcheck currently says:

```
In foo.sh line 2:
find . -type f -print0 | xargs --null cat -- | nice -15 cat
                                               ^-- SC2216: Piping to 'nice', a command that doesn't read stdin. Wrong command or missing xargs?
```

#### Here's what I wanted or expected to see:

Nothing.

#### Explanation

The 'nice' command runs other commands, so piping to 'nice' is fine when it is running a command and that command is not known to be one that doesn't read stdin.

**Comment by koalaman** [comment]
You're right, of course. This embarrassing issue was fixed for #1169
